server/controllers/booking.js

```diff
@@ -0,0 +1,64 @@
+const Booking = require("../models/Booking");
+const Profile = require("../models/Profile");
+const asyncHandler = require("express-async-handler");
+
+// @route GET /bookings
+// @desc list of bookings
+// @access Private
+exports.getBookings = asyncHandler(async (req, res, next) => {
+  const profile = await Profile.findOne({ userId: req.user.id });
+  let bookings;
+  if (profile.accountType === "pet_owner") {
+    bookings = await Booking.find({ petOwner: req.user.id }).populate([
+      "petOwner",
+      "sitter",
+    ]);
+  } else {
+    bookings = await Booking.find({ sitter: req.user.id }).populate([
+      "petOwner",
+      "sitter",
+    ]);
+  }
+
+  res.status(200).json({
+    success: {
+      bookingInfos: bookings,
+    },
+  });
+});
+
+// @route POST /bookings
+// @desc Create a new booking
+// @access Private
+exports.makeBooking = asyncHandler(async (req, res, next) => {
```

Comment:
```suggestion
exports.makeBooking = asyncHandler(async (req, res, next) => {
  const {sitter, startDate, endDate} = req.booking
  const booking = new Booking({
    sitter,
    startDate,
    endDate,
    petOwner: req.user.id
  });
  
 const newBooking = await Booking.create(booking);

  res.status(200).json({
    success: {
      bookingInfo: newBooking,
    },
  });
});
```
We should destructure the request body to only allow selected fields (for security purposes). Also, we want to supply the logged is user's id from the request object rather than sending it in the request body